api/base/serializers.py

```diff
@@ -80,6 +81,22 @@ def to_internal_value(self, data):
         return super(TypeField, self).to_internal_value(data)
 
 
+class TargetTypeField(ser.CharField):
```

Comment:
I was originally subclassing TypeField, but couldn't call `super` on TargetTypeField because in addition to checking `target_type` against `target_type`, it would check `target_type` against `type`.  Since I was calling super on a different parent, it was suggested that I subclass CharField.  Avoiding the weird super call was worth a few lines of duplicate code.
https://github.com/CenterForOpenScience/osf.io/pull/4439#discussion_r42504859